x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/inference/textembedding/TextEmbeddingOperatorRequestIteratorTests.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.inference.textembedding;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.compute.data.BytesRefBlock;
+import org.elasticsearch.compute.test.ComputeTestCase;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xpack.core.inference.action.InferenceAction;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class TextEmbeddingOperatorRequestIteratorTests extends ComputeTestCase {
+
+    public void testIterateSmallInput() throws Exception {
+        assertIterate(between(1, 100));
+    }
+
+    public void testIterateLargeInput() throws Exception {
+        assertIterate(between(10_000, 100_000));
+    }
+
+    public void testIterateWithNullValues() throws Exception {
+        final String inferenceId = randomIdentifier();
+        final BytesRefBlock inputBlock = createBlockWithNulls();
+
+        try (TextEmbeddingOperatorRequestIterator requestIterator = new TextEmbeddingOperatorRequestIterator(inputBlock, inferenceId)) {
+            BytesRef scratch = new BytesRef();
+
+            // First position: "before"
+            InferenceAction.Request request1 = requestIterator.next();
+            assertThat(request1.getInferenceEntityId(), equalTo(inferenceId));
+            assertThat(request1.getTaskType(), equalTo(TaskType.TEXT_EMBEDDING));
+            scratch = inputBlock.getBytesRef(inputBlock.getFirstValueIndex(0), scratch);
+            assertThat(request1.getInput().get(0), equalTo(scratch.utf8ToString()));
+
+            // Second position: null
+            InferenceAction.Request request2 = requestIterator.next();
+            assertThat(request2, nullValue());
+
+            // Third position: "after"
+            InferenceAction.Request request3 = requestIterator.next();
+            assertThat(request3.getInferenceEntityId(), equalTo(inferenceId));
+            assertThat(request3.getTaskType(), equalTo(TaskType.TEXT_EMBEDDING));
+            scratch = inputBlock.getBytesRef(inputBlock.getFirstValueIndex(2), scratch);
+            assertThat(request3.getInput().get(0), equalTo(scratch.utf8ToString()));
+        }
+
+        allBreakersEmpty();
+    }
+
+    public void testIterateWithMultiValuePositions() throws Exception {
+        final String inferenceId = randomIdentifier();
+        final BytesRefBlock inputBlock = createMultiValueBlock();
+
+        try (TextEmbeddingOperatorRequestIterator requestIterator = new TextEmbeddingOperatorRequestIterator(inputBlock, inferenceId)) {
+            // First position: multi-value concatenated with newlines
```

Comment:
I'm not sure if we're testing what the comment says? I don't see a check for the multivalues 🤔